src/test/java/com/sap/sgs/phosphor/fosstars/data/github/GitHubDataFetcherTest.java

```diff
@@ -79,4 +92,349 @@ public void testShouldUpdate() throws IOException {
+          latch.countDown();
+        });
+      }
+      latch.await();
+
+      checkCleanUp(project, 0);
+    } finally {
+      executorService.shutdown();
+    }
+  }
+
+  @Test
+  public void testLocalRepositoriesInfoJson() throws IOException {
+    // Create the first project with first GitHub mock instance.
+    GitHubProject firstProject = new GitHubProject("test", "project1");
+    Path project1Dir = projectDir(firstProject.name());
+    GitHub github = mock(GitHub.class);
+    try (Repository repository =
+        FileRepositoryBuilder.create(project1Dir.resolve(".git").toFile())) {
+      repository.create();
+
+      createRepoTestFile(repository, firstProject, "testFile1");
+
+      fetcher = spy(new TestGitHubDataFetcher(github));
+
+      fetcher.addRepositoryInfoForTesting(firstProject, project1Dir);
+
+      localRepositoryFor(firstProject, 1);
+    }
+
+    // Create the second project with second GitHub mock instance.
+    GitHubProject secondProject = new GitHubProject("test", "project2");
+    Path project2Dir = projectDir(secondProject.name());
+    try (Repository repository =
+        FileRepositoryBuilder.create(project2Dir.resolve(".git").toFile())) {
+      repository.create();
+
+      createRepoTestFile(repository, secondProject, "testFile2");
+
+      fetcher.addRepositoryInfoForTesting(secondProject, project2Dir);
+
+      localRepositoryFor(secondProject, 2);
+    }
+
+    cleanup(firstProject, 1);
+    cleanup(secondProject, 0);
+  }
+
+  @Test
+  public void testMultipleInstanceFetcher() throws IOException {
+    // Create the first project with first GitHub mock instance.
+    GitHubProject firstProject = new GitHubProject("test", "project1");
+    Path project1Dir = projectDir(firstProject.name());
+    GitHub github = mock(GitHub.class);
+    try (Repository repository =
+        FileRepositoryBuilder.create(project1Dir.resolve(".git").toFile())) {
+      repository.create();
+
+      createRepoTestFile(repository, firstProject, "testFile1");
+
+      fetcher = spy(new TestGitHubDataFetcher(github));
+      
+      fetcher.addRepositoryInfoForTesting(firstProject, project1Dir);
+
+      localRepositoryFor(firstProject, 1);
+    }
+
+    // Create the second project with second GitHub mock instance.
+    GitHubProject secondProject = new GitHubProject("test", "project2");
+    Path project2Dir = projectDir(secondProject.name());
+    try (Repository repository =
+        FileRepositoryBuilder.create(project2Dir.resolve(".git").toFile())) {
+      repository.create();
+
+      createRepoTestFile(repository, secondProject, "testFile2");
+      
+      fetcher = spy(new TestGitHubDataFetcher(github));
+
+      fetcher.addRepositoryInfoForTesting(secondProject, project2Dir);
+
+      localRepositoryFor(secondProject, 2);
+    }
+  }
+
+  @Test
+  public void testLocalRepsitoriesCapacity() throws IOException {
+    GitHub github = mock(GitHub.class);
+    fetcher = spy(new TestGitHubDataFetcher(github));
```

Comment:
I am not sure why you always wrap it with `spy()`.

Reply:
Not required